### ISSUE TITLE
How long does it take me to train for one epoch?

### ISSUE THREAD

**Issue by Disyxt** [ml-struct-bio/drgnai#26]
I have been training for several hours, but train.rog has not updated the training data. I don't know if it's a configuration issue or if the training should have taken so long?

**Comment by michal-g** [comment]
Sorry to hear you are having trouble — just to clarify, you are referring to the log file (`training.log`) saved to your model output folder during training? Is this file still empty, or does it contain any logs before it stopped updating?

Can you also provide some information about the training configurations you were using and the size of your dataset (especially # of particles and resolution)?

**Comment by Disyxt** [comment]
Thank you for your response.
My particle count is about 140000. The pixel size is 640。my log is look like：
----------
✅ CUDA cache cleared
CUDA available: True
Device count: 3
Device name: NVIDIA A40
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Using existing output directory which does not yet contain any drgnai output!.
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Using 3 GPUs!
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Increasing batch size for known poses to 48
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Increasing batch size for HPS to 24
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Increasing batch size for SGD to 96
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Use cuda True
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Will write tensorboard summaries in test/out/summaries
(INFO) (reconstruct.py) (22-Oct-25 10:34:46) Creating dataset
(INFO) (dataset.py) (22-Oct-25 10:34:46) [INFO] Lazy loading mode = True
[INFO] Lazy mode enabled when loading MRC particles
[INFO] Using lazy loading for CryoSPARC .cs file (no preloading)
[INFO] Using lazy sample loading for normalization (n=50)
(INFO) (dataset.py) (22-Oct-25 10:34:52) Lazy loaded 131642 640x640 images
(INFO) (dataset.py) (22-Oct-25 10:34:52) Windowing images with radius 0.85
(INFO) (dataset.py) (22-Oct-25 10:34:52) Spawning 16 processes
(INFO) (reconstruct.py) (22-Oct-25 10:34:53) Loading ctf params from /work/home/xiaoting/drgnai_slp1/inputs2/ctf.pkl
(INFO) (ctf.py) (22-Oct-25 10:34:53) Image size (pix)  : 640
(INFO) (ctf.py) (22-Oct-25 10:34:53) A/pix             : 1.0700000524520874
(INFO) (ctf.py) (22-Oct-25 10:34:53) DefocusU (A)      : 4070.08642578125
(INFO) (ctf.py) (22-Oct-25 10:34:53) DefocusV (A)      : 3929.3486328125
(INFO) (ctf.py) (22-Oct-25 10:34:53) Dfang (deg)       : -27.306190490722656
(INFO) (ctf.py) (22-Oct-25 10:34:53) voltage (kV)      : 300.0
(INFO) (ctf.py) (22-Oct-25 10:34:53) cs (mm)           : 2.700000047683716
(INFO) (ctf.py) (22-Oct-25 10:34:53) w                 : 0.10000000149011612
(INFO) (ctf.py) (22-Oct-25 10:34:53) Phase shift (deg) : 0.0
(INFO) (reconstruct.py) (22-Oct-25 10:34:53) Building lattice
(INFO) (reconstruct.py) (22-Oct-25 10:34:53) Heterogeneous reconstruction with z_dim = 4
(INFO) (reconstruct.py) (22-Oct-25 10:34:53) Initializing model...
(INFO) (reconstruct.py) (22-Oct-25 10:34:53) DrgnAI(
  (pose_table): PoseTable()
  (conf_table): ConfTable()
  (hypervolume): HyperVolume(
    (mlp): ResidualLinearMLP(
      (main): Sequential(
        (0): Linear(in_features=388, out_features=256, bias=True)
        (1): ReLU()
        (2): ResidualLinear(
          (linear): Linear(in_features=256, out_features=256, bias=True)
        )
        (3): ReLU()
        (4): ResidualLinear(
          (linear): Linear(in_features=256, out_features=256, bias=True)
        )
        (5): ReLU()
        (6): ResidualLinear(
          (linear): Linear(in_features=256, out_features=256, bias=True)
        )
        (7): ReLU()
        (8): MyLinear(in_features=256, out_features=1, bias=True)
      )
    )
  )
)
(INFO) (reconstruct.py) (22-Oct-25 10:34:53) 1876921 parameters in model
(INFO) (reconstruct.py) (22-Oct-25 10:34:53) Model initialized. Moving to GPU...
(WARNING) (reconstruct.py) (22-Oct-25 10:34:58) torch.amp mixed precision training is not optimized; self.configs.z_dim=4 is not a multiple of 8!
(INFO) (reconstruct.py) (22-Oct-25 10:34:58) Using Automatic Mixed Precision training via torch.amp
(INFO) (reconstruct.py) (22-Oct-25 10:34:58) --- Training Starts Now ---
(INFO) (reconstruct.py) (22-Oct-25 10:34:58) Will pretrain on 10000 particles
(INFO) (reconstruct.py) (22-Oct-25 10:34:58) Will make a full summary at the end of this epoch
------------
It's been over eight hours and there's still no update。
I also test the same data whlie pixel size is 160，but the situation is the same。

**Comment by Disyxt** [comment]
Thanks! I have already solved this problem. After adding the following lines of code, I successfully started running! And I have lowered the batch_size because my GPU is A40.

export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True,max_split_size_mb:64
export NCCL_P2P_DISABLE=1
export NCCL_IB_DISABLE=1
export NCCL_DEBUG=INFO
----
